clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -145,7 +155,16 @@ private SSLContext createSSLContext() throws GeneralSecurityException, IOExcepti
         KeyStore ts = truststore == null ? null : truststore.load();
         tmf.init(ts);
 
-        sslContext.init(keyManagers, tmf.getTrustManagers(), this.secureRandomImplementation);
+        TrustManager[] trustManagers = tmf.getTrustManagers();
+        TrustManager reloadableTrustManager = new ReloadableX509TrustManager(truststore, tmf);
```

Comment:
I have update the pull request to use reloadable trust manager when a configured truststore is used, client-auth is used and trust manager is used as server mode.